web/src/components/njwds-extended/Heading.tsx

```diff
@@ -0,0 +1,110 @@
+import { ReactElement } from "react";
+
+export type ValidHeadingLevel = 0 | 1 | 2 | 3 | 4;
+export type HeadingStyleVariantName = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "rawElement";
+
+type HeadingStyles = {
+  [variantName in HeadingStyleVariantName]: string;
+};
+
+const variantClasses: HeadingStyles = {
+  h1: "h1-styling",
+  h2: "h2-styling",
+  h3: "h3-styling",
+  h4: "h4-styling",
+  h5: "h5-styling",
+  h6: "h6-styling",
+  rawElement: "",
+};
+
+const determineClasses = (
+  level: ValidHeadingLevel,
+  styleVariant?: HeadingStyleVariantName | undefined,
+  className?: string | undefined
+): string => {
+  const classStrings = [];
+  let styleClasses: string;
```

Comment:
So, here's my thinking, and I'd love your feedback on this since I still haven't learned how much ahead-of-time planning we generally do.

The thought here is that, while the `variantClasses` currently only have 0-1 class each, these could be expanded in the future to contain multiple classes to apply to all h2 styles, h3 styles, etc.

Reply:
I see. I don't think we need to plan in advance in this case. I think we should have the variable name reflect the current condition, as not to cause confusion and to also allow the name to speak to what the is happening in the code at the moment. Should a future case call for multiple classes applying to an `h` tag, it's an easy enough change and we can update the variable name as we see fit then.

Reply:
Weak opinion, weakly-held, of course. Totally open to changing the variable name, but I wanted to solicit your feedback before doing so.